### ISSUE TITLE
Footnote content with hyperlink breaks DOM tab order

### ISSUE THREAD

**Issue by GarthMartin** [wet-boew/wet-boew#9530]
## Required information

If the content of a Footnote contains a hyperlink, the TAB order of the page no longer matches the DOM because the footnote number is pulled to the left of the hyperlink content.

**Expected behaviour**
Tab order should be like this:
![image](https://user-images.githubusercontent.com/11050106/219129123-293dd0e2-fdc1-4c41-a4e4-e1df83fbe9c2.png)

But it's like this:
![image](https://user-images.githubusercontent.com/11050106/219129210-a3377c24-e97a-48cb-bb2a-6c75c5c7204b.png)

**Version of WET-BOEW/GCWeb you are using**
e.g. v4.0.43

**Comment by acev0010** [comment]
Hello,

Could you please provide the page where you found this issue, or did this happen in a local environment?

**Comment by EricDunsworth** [comment]
(**Full disclosure:** I might be biased since I created WET's footnotes plugin 😝)

FWIW the footnotes plugin was deliberately designed in this way.

I believe this issue relates to success criteria [1.3.2 (meaningful sequence)](https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence.html) and [2.4.3 (focus order)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html). Here are my takes on them...

### [SC 1.3.2 (meaningful sequence)](https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence.html):
I'd consider it to pass since IMO it follows techniques [G57](https://www.w3.org/WAI/WCAG21/Techniques/general/G57) + [C6](https://www.w3.org/WAI/WCAG21/Techniques/css/C6).

The structural order makes sense to me (``<dt>`` elements for footnote identifiers, followed by ``<dd>`` elements for footnote content and return links at the end). CSS is used to visually-hide the ``<dt>``s and visually-reposition the return links to present them as footnote identifiers + return links. The CSS presentation corresponds to how footnotes are usually represented in print media.

As a user, I'd also consider the status quo to be the an effective way of tabbing through footnotes since it goes through footnote content first, followed by the return link. Most users would probably prefer this over the alternative.

If the DOM structure were to be reworked to match the visual order, it might betray user expectations. Why? Because the return links would now come before the footnotes themselves - depriving users of the chance of tabbing through each footnote's interactive elements before being given a means of returning to the main content.

I guess another way of doing it could be to unveil the ``<dt>`` elements' footnote identifiers on the left and show the return links in their original positions (as paragraphs at the end of each footnote... or some combination of icons and shortened text). But that'd probably look bloated and feel like a chore for sighted users to use vs the status quo 😟.

### [SC 2.4.3 (focus order)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html):
I'd consider this to pass too via [technique G59](https://www.w3.org/WAI/WCAG21/Techniques/general/G59).

Even though the focus order isn't from left-to-right, IMO it still follows a logical sequence that preserves meaning and operability. The only twist is that it's based on structural markup as opposed to visual representation.

Both the understanding and technique pages contain "outs" for this kind of scenario:
* [Excerpt from SC 2.4.3's understanding page](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html):
  > **The focus order may not be identical to the programmatically determined reading order (see Success Criterion 1.3.2) as long as the user can still understand and operate the Web page.** Since there may be several possible logical reading orders for the content, the focus order may match any of them. However, when the order of a particular presentation differs from the programmatically determined reading order, users of one of these presentations may find it difficult to understand or operate the Web page. Authors should carefully consider all these users as they design their Web pages. 
* [Excerpt from technique G59](https://www.w3.org/WAI/WCAG21/Techniques/general/G59):
  > As an example, in HTML, the default focus order follows the order in which elements appear in the content source. When the order of the HTML source matches the visual order of the Web page, tabbing through the content follows the visual layout of the content. **When the source order does not match the visual order, the tab order through the content must reflect the logical relationships in the content that are displayed visually.**

### PS:
All in all, I don't think anything's fundamentally wrong with the sequence/focus order. But I will concede that the status quo might not be doing enough to make the purpose of the return links clear to sighted users.

As things stand, footnote reference links look like small numbers and return links look like normal-sized numbers. It's easy for most users to get the hang of... but only after clicking the links for the first time.

IMO it'd be better if the purposes of both kinds of links (especially return ones) were visually-obvious without having to click on them in the first place to "learn" how they work. But I was unable to come up with a way of doing it that doesn't look overly-bloated or radically different from how footnotes are traditionally represented in print media.

---

@acev0010 I wasn't able to find the page @GarthMartin's screenshots were taken from, but it should be possible to reproduce that scenario by adding a link into any footnote's content in any implementation of this plugin.

If you have a cloned version of the WET repo and a working build system, the easiest way would probably be to modify your version of [``src/plugins/footnotes/footnotes-en.hbs``](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/footnotes/footnotes-en.hbs), run ``grunt`` and check your unminified version of the [footnotes demo page](https://wet-boew.github.io/wet-boew/unmin/demos/footnotes/footnotes-en.html).

**Comment by EricDunsworth** [comment]
@acev0010

To check your unmodified version of the footnotes demo page in a web browser, I'd recommend the following steps:
1. Run ``grunt`` ("compiles" the source files into a ``/dist/`` folder)
2. Run ``grunt server`` (should show something along the lines of "Started connect web server on [``http://localhost:8000``](http://localhost:8000)")
3. Visit your [compiled version of the footnotes demo page](http://localhost:8000/unmin/demos/footnotes/footnotes-en.html) in any web browser.

To edit your copy of the footnotes demo page:
1. Leave ``grunt server`` running
2. Open the footnotes demo's source file ([src/plugins/footnotes/footnotes-en.hbs](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/footnotes/footnotes-en.hbs)) in a text editor
3. Edit it to your liking (like adding an extra link to footnote 1's example content)
4. ``grunt server`` will automatically detect that the file changed once you save it... wait a moment for it to recompile
5. Refresh your [compiled version of the footnotes demo page](http://localhost:8000/unmin/demos/footnotes/footnotes-en.html) in a web browser

That's the simplest way I can think of to do it without setting up your own local web server. If you find yourself making many small edits to your copy of the demo page and can't stand waiting for ``grunt server`` to constantly recompile every time you save, your best bet would be to edit the compiled version (``/dist/unmin/demos/footnotes/footnotes-en.html``) directly.

Btw you can run ``grunt -h`` to see an overview of all available grunt commands in WET's build system.

Hope this helps :)!
